### PR TITLE
Tidy/Reorganise autoscan files

### PR DESCRIPTION
**Pull request by whyman**

### CODE REVIEW

File: src/content/content_manager.cc [hunk 1]
```diff
@@ -659,7 +659,7 @@ void ContentManager::_rescanDirectory(const std::shared_ptr<AutoscanDirectory>&
 
         // it is possible that someone hits remove while the container is being scanned
         // in this case we will invalidate the autoscan entry
-        if (adir->getScanID() == INVALID_SCAN_ID) {
+        if (adir->getScanID() == AutoscanDirectory::INVALID_SCAN_ID) {
```
Comment (KarlStraussberger) [inline src/content/content_manager.cc:662]: Encapsulate as `adir->IsInvalid()`?

File: src/database/sql_database.cc [hunk 2]
```diff
@@ -277,7 +277,7 @@ static const std::vector<std::pair<std::string, MetadataCol>> metaTagMap {
 static const std::vector<std::pair<std::string, AutoscanCol>> asTagMap {
     { "id", AutoscanCol::Id },
     { "obj_id", AutoscanCol::ObjId },
-    { "persistent", AutoscanCol::Persistent },
+    { "isPersistent", AutoscanCol::Persistent },
```
Comment (KarlStraussberger) [inline src/database/sql_database.cc:280]: That's the column name in the database!

File: src/database/sql_database.cc [hunk 3]
```diff
@@ -292,7 +292,7 @@ static const std::vector<std::pair<std::string, AutoscanColumn>> autoscanTagMap
     { "hidden", AutoscanColumn::Hidden },
     { "interval", AutoscanColumn::Interval },
     { "last_modified", AutoscanColumn::LastModified },
-    { "persistent", AutoscanColumn::Persistent },
+    { "isPersistent", AutoscanColumn::Persistent },
```
Comment (KarlStraussberger) [inline src/database/sql_database.cc:295]: That's the column name as well

File: src/database/sql_database.cc [hunk 4]
```diff
@@ -2174,19 +2174,19 @@ void SQLDatabase::addAutoscanDirectory(const std::shared_ptr<AutoscanDirectory>&
         identifier("hidden"),
         identifier("interval"),
         identifier("last_modified"),
-        identifier("persistent"),
+        identifier("isPersistent"),
```
Comment (KarlStraussberger) [inline src/database/sql_database.cc:2177]: Column name again

File: src/server.h [hunk 5]
```diff
@@ -112,7 +112,7 @@ class Server : public std::enable_shared_from_this<Server> {
     /// \brief Unique Device Number of the server.
     ///
     /// The UDN is read from the config, it must be unique and
-    /// persistent over reboots.
+    /// isPersistent over reboots.
```
Comment (KarlStraussberger) [inline src/server.h:115]: Ooops

File: src/web/directories.cc [hunk 6]
```diff
@@ -101,13 +101,13 @@ void Web::Directories::process()
         ce.append_attribute("child_count") = has;
         auto aDir = std::find_if(autoscanDirs.begin(), autoscanDirs.end(), [&](auto& a) { return file == a->getLocation(); });
         if (aDir != autoscanDirs.end()) {
-            ce.append_attribute("autoscan_type") = (*aDir)->persistent() ? "persistent" : "ui";
-            ce.append_attribute("autoscan_mode") = AutoscanDirectory::mapScanmode((*aDir)->getScanMode());
+            ce.append_attribute("autoscan_type") = (*aDir)->isPersistent() ? "isPersistent" : "ui";
```
Comment (KarlStraussberger) [inline src/web/directories.cc:104]: That's used in JavaScript

File: src/web/web_autoscan.cc [hunk 7]
```diff
@@ -138,12 +138,12 @@ void Web::Autoscan::autoscan2XML(const std::shared_ptr<AutoscanDirectory>& adir,
         element.append_child("recursive").append_child(pugi::node_pcdata).set_value("0");
         element.append_child("hidden").append_child(pugi::node_pcdata).set_value("0");
         element.append_child("interval").append_child(pugi::node_pcdata).set_value("1800");
-        element.append_child("persistent").append_child(pugi::node_pcdata).set_value("0");
+        element.append_child("isPersistent").append_child(pugi::node_pcdata).set_value("0");
```
Comment (KarlStraussberger) [inline src/web/web_autoscan.cc:141]: Javascript again

File: src/web/web_autoscan.cc [hunk 8]
```diff
@@ -138,12 +138,12 @@ void Web::Autoscan::autoscan2XML(const std::shared_ptr<AutoscanDirectory>& adir,
...
     } else {
-        element.append_child("scan_mode").append_child(pugi::node_pcdata).set_value(AutoscanDirectory::mapScanmode(adir->getScanMode()));
-        element.append_child("recursive").append_child(pugi::node_pcdata).set_value(adir->getRecursive() ? "1" : "0");
+        element.append_child("scan_mode").append_child(pugi::node_pcdata).set_value(AutoscanDirectory::mapScanMode(adir->getScanMode()));
+        element.append_child("recursive").append_child(pugi::node_pcdata).set_value(adir->isRecursive() ? "1" : "0");
         element.append_child("hidden").append_child(pugi::node_pcdata).set_value(adir->getHidden() ? "1" : "0");
         element.append_child("interval").append_child(pugi::node_pcdata).set_value(fmt::to_string(adir->getInterval().count()).c_str());
-        element.append_child("persistent").append_child(pugi::node_pcdata).set_value(adir->persistent() ? "1" : "0");
+        element.append_child("isPersistent").append_child(pugi::node_pcdata).set_value(adir->isPersistent() ? "1" : "0");
```
Comment (KarlStraussberger) [inline src/web/web_autoscan.cc:147]: JavaScript

File: src/content/content_manager.cc (hunk 1 above)
Comment (whyman) [reply src/content/content_manager.cc:662]: great idea!

File: src/database/sql_database.cc (hunk 2 above)
Comment (whyman) [reply src/database/sql_database.cc:280]: Editor was a bit overzealous

**Comment by whyman** [comment]
Closing this, going to incorporate it into a larger change

**Pull request closed without merging**